content/guides/typescript-for-solid.mdx

```diff
@@ -0,0 +1,714 @@
+
+```typescript
+import { type Component } from "solid-js";
+
+function MyTsComponent(): JSX.Element {
+  return (
+    <div>
+      <h1>This is a TypeScript component</h1>
+    </div>
+  );
+}
+
+export default MyTsComponent;
+```
+
+To use in an existing JavaScript component, import the TypeScript component:
+
+```javascript
+import MyTsComponent from "./MyTsComponent";
+
+function MyJsComponent() {
+  return (
+    <>
+      {/* ... */}
+      <MyTsComponent />
+    </>
+  );
+}
+```
+
+<Aside>
+**Note:** If you wish to change the entry point file from `index.jsx` to `index.tsx`, you need to modify the `src` attribute in `<script>` to look like the following:
+
+```htmlembedded
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <!-- ... -->
+  </head>
+  <body>
+    <noscript>You need to enable JavaScript to run this app.</noscript>
+    <div id="root"></div>
+    <script src="/src/index.tsx" type="module"></script>
+  </body>
+</html>
+```
+</Aside>
+
+## API types
+
+Solid is written in TypeScript, meaning everything is typed out of the box.
+
+The [API Documentation](/references) details the types of all the API calls. 
+In addition, there are several helpful type definitions to make it easier when you need to specify explicit types.
+
+### Signals
+
+Using `createSignal<T>`, you can define the signal's type as `T`. 
+
+```typescript
+const [count, setCount] = createSignal<number>();
+```
+
+Here, `createSignal` has the return type `Signal<number | undefined>`, corresponding to the type passed into it, and `undefined` as it was uninitialized. 
+This resolves to a getter-setter [tuple](https://www.typescriptlang.org/docs/handbook/2/objects.html#tuple-types), both of which are generically typed:
+
+```typescript
+import type { Signal, Accessor, Setter } from "solid-js";
+type Signal<T> = [get: Accessor<T>, set: Setter<T>];
+```
+
+In this case, the `count` signal getter has the type `Accessor<number | undefined>`. 
+In Solid, `Accessor<T>` is equivalent to `() => T`. 
+Here, it means `count` is a function that returns either a `number` or `undefined`.
+
+The reason `undefined` is included in the type is because the signal was not initialized. 
+Therefore, its initial state can be `undefined`.
+
+The signal setter, `setCount`, is of type `Setter<number | undefined>`. 
+This complex type roughly resembles `(value? : number | ((prev?: number) => number)) => number`, accommodating both a direct number or a function which takes a potential pervious value that results in a number.
+Since the signal's start is undefined, both the direct number and the function's number parameter are optional.
+
+When a signal is designed to store functions, its setter function won't directly accept new functions as values.
+This is because the setter can't determine whether the function should be executed to get the actual value. 
+In these situations, you should use the callback form of the setter: `setSignal(() => value)`.
```

Comment:
```suggestion
In Solid, a signal's getter, like `count`, is essentially a function that returns a specific type. 
In this case, the type is `Accessor<number | undefined>`, which translates to a function `() => number | undefined`. 
Including `undefined` in the type indicates that the signal may not have been initialized, so its initial state could be `undefined`.

The corresponding setter, `setCount`, has a more complex type: `Setter<number | undefined>`. 
This type can be roughly understood as a function that takes either a direct number or another function as its argument. 
This function argument, in turn, can take the previous value of the signal and return a new value. 
Both the input and output can be either a number or `undefined`. 
Notably, invoking `setCount` without any parameters will set the signal's value to `undefined`.

When using the function form of the setter, a single argument is provided, which can be ignored if desired.
Additionally, the return type of the setter will align with the type of value passed into it, echoing the behavior expected from a typical assignment operation.

If a signal is intended to store functions, the setter won't directly accept new functions as values. 
This is because it can not distinguish whether the function should be executed to yield the actual value or stored as is. 
In these situations, using the callback form of the setter is recommended: `setSignal(() => value)`.
```

@otonashixav based on what you mentioned above, I rewrote this part. Let me know what you think

Reply:
```diff
- Including `undefined` in the type indicates that the signal may not have been initialized, so its initial state could be `undefined`.
+ As the signal was not initialized, its initial state is `undefined`, therefore `undefined` is included in its type.
```

Otherwise looks good.